contrib/proguard/src/Proguard.scala

```diff
@@ -0,0 +1,84 @@
+package mill.contrib.proguard
+
+import coursier.Repositories
+import mill.T
+import mill.Agg
+import mill.api.{Logger, Loose, PathRef, Result}
+import mill.define.{Sources, Target}
+import mill.scalalib.Lib.resolveDependencies
+import mill.scalalib.{Dep, DepSyntax, Lib, ScalaModule}
+import os.proc
+import os.Path
+import os.PathChunk
+
+trait Proguard extends ScalaModule {
+
+  // This is actually the only available version on Maven currently.
+  // But we should keep it as a configuration in case a new version is released.
+  def proguardVersion: T[String] = T { "7.0.0" }
+
+  def shrink: T[Boolean] = T { true }
+  def optimize: T[Boolean] = T { true }
+  def obfuscate: T[Boolean] = T { true }
+  def preverify: T[Boolean] = T { true }
+
+  def javaHome: T[PathRef] = T {
+    PathRef(Path(System.getProperty("java.home")))
+  }
+
+  def inJar: T[PathRef] = T { assembly() }
+
+  def libraryJars: T[Seq[PathRef]] = T {
+    val javaJars = os.list(javaHome().path / "lib", sort = false).filter(_.ext == "jar")
+    javaJars.toSeq.map(PathRef(_))
+  }
+
+  def proguard: T[PathRef] = T {
+    val outJar = PathRef(T.dest / "out.jar")
+
+    val cmd = os.proc(
+      "java",
```

Comment:
Better use the `java` executable found under `javaHome`, to make it consistent with the later used classpath.